core/services/bulletprooftxmanager/eth_confirmer_test.go

```diff
@@ -954,6 +954,112 @@ func TestEthConfirmer_BumpGasWhereNecessary(t *testing.T) {
 	ethClient.AssertExpectations(t)
 }
 
+func TestEthConfirmer_BumpGasWhereNecessary_WhenOutOfEth(t *testing.T) {
+	t.Parallel()
+
+	store, cleanup := cltest.NewStore(t)
+	defer cleanup()
+	ethClient := new(mocks.Client)
+	store.EthClient = ethClient
+
+	// Use the real KeyStore loaded from database fixtures
+	store.KeyStore.Unlock(cltest.Password)
+
+	config, cleanup := cltest.NewConfig(t)
+	defer cleanup()
+
+	ec := bulletprooftxmanager.NewEthConfirmer(store, config)
+	currentHead := int64(30)
+	oldEnough := int64(19)
+	nonce := int64(0)
+
+	etx := cltest.MustInsertUnconfirmedEthTxWithBroadcastAttempt(t, store, nonce)
+	nonce++
+	attempt1_1 := etx.EthTxAttempts[0]
+	attempt1_1.BroadcastBeforeBlockNum = &oldEnough
+	require.NoError(t, store.DB.Save(&attempt1_1).Error)
+	var attempt1_2 models.EthTxAttempt
+	var err error
+
+	insufficientEthError := errors.New("insufficient funds for gas * price + value")
+
+	t.Run("saves attempt with state 'insufficient_eth' if eth node returns this error", func(t *testing.T) {
+		expectedBumpedGasPrice := big.NewInt(25000000000)
+		require.Greater(t, expectedBumpedGasPrice.Int64(), attempt1_1.GasPrice.ToInt().Int64())
+
+		ethClient.On("SendTransaction", mock.Anything, mock.MatchedBy(func(tx *gethTypes.Transaction) bool {
+			return expectedBumpedGasPrice.Cmp(tx.GasPrice()) == 0
+		})).Return(insufficientEthError).Once()
+
+		// Do the thing
```

Comment:
:scissors: or make more descriptive

Reply:
This comment is already everywhere :D it's just to help find the line that actually execute the code under test.

Can you suggest a better phrase?

Reply:
haha these are everywhere :man_facepalming: guess I'm too late to make the request